sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/Preprocessor.scala

```diff
@@ -202,6 +208,16 @@ private[engine] final class Preprocessor(
       } yield m.updated(pkgName, pkgId)
     )
 
+  def translateKey(
+      templateId: Ref.TypeConName,
+      contractKey: Value,
+  ): Result[speedy.SValue] = {
+    safelyRun(pullPackage(Seq(templateId))) {
+      val ckType = handleLookup(pkgInterface.lookupTemplateKey(templateId)).typ
+      valueTranslator.unsafeTranslateValue(ty = ckType, upgradable = true, value = contractKey)
```

Comment:
It's not obvious to me why `upgradable = true` is a safe choice. The `CommandPreprocessor` would use `this.upgradable(templateId)` for this field here.